staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -328,52 +363,56 @@ func (p *jsonPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (r
 	return objToUpdate, nil
 }
 
-func (p *jsonPatcher) createNewObject() (runtime.Object, error) {
+func (p *jsonPatcher) createNewObject(_ context.Context) (runtime.Object, error) {
 	return nil, errors.NewNotFound(p.resource.GroupResource(), p.name)
 }
 
 // applyJSPatch applies the patch. Input and output objects must both have
 // the external version, since that is what the patch must have been constructed against.
-func (p *jsonPatcher) applyJSPatch(versionedJS []byte) (patchedJS []byte, retErr error) {
+func (p *jsonPatcher) applyJSPatch(versionedJS []byte) (patchedJS []byte, strictErrors []error, retErr error) {
 	switch p.patchType {
 	case types.JSONPatchType:
 		// sanity check potentially abusive patches
 		// TODO(liggitt): drop this once golang json parser limits stack depth (https://github.com/golang/go/issues/31789)
```

Comment:
the issue is resolved in go1.15+... can drop this block for that purpose, rather than augmenting it for this purpose

it looks like the code you added will catch duplicate fields, but nothing else... we also want to catch unknown fields in json patches (the only fields in the top level of each item should be "op", "path", "from", and "value"), so switch this to `[]jsonPatchOp`, where jsonPatchOp has those four fields defined (also need test coverage)